analyzers/src/SonarAnalyzer.Common/Rules/Utilities/AnalysisWarningAnalyzerBase.cs

```diff
@@ -0,0 +1,61 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.IO;
+using SonarAnalyzer.CFG.Helpers;
+
+namespace SonarAnalyzer.Rules;
+
+public abstract class AnalysisWarningAnalyzerBase : UtilityAnalyzerBase
+{
+    private const string DiagnosticId = "S9999-warning";
+    private const string Title = "Analysis Warning generator";
+
+    private static readonly object FileWriteLock = new();
+
+    protected virtual int MinimalSupportedRoslynVersion { get; } = RoslynHelper.MinimalSupportedMajorVersion;   // For testing
+
+    protected AnalysisWarningAnalyzerBase() : base(DiagnosticId, Title) { }
+
+    protected sealed override void Initialize(SonarAnalysisContext context) =>
+        context.RegisterCompilationAction(c =>
+            {
+                ReadParameters(c);
+                if (IsAnalyzerEnabled && !RoslynHelper.IsRoslynCfgSupported(MinimalSupportedRoslynVersion))     // MsBuild 15 is bound with Roslyn 2.x, where Roslyn CFG is not available.
```

Comment:
I think that `RoslynHelper.MinimalSupportedMajorVersion` should be private in `RoslynHelper`. `AnalysisWarningAnalyzer` does not need to know about it and it can be encapsulated.

Reply:
It would be nicer to be private. But I need it for testing. And I don't want to create heavy indirection mechanism for this rather temporary utility analyzer.